constraint/pkg/client/client.go

```diff
@@ -377,6 +382,51 @@ func (c *Client) AddConstraint(ctx context.Context, constraint *unstructured.Uns
 	return resp, nil
 }
 
+// withDefaultParams injects any default values defined in the OpenAPIV3Schema
+// to the constraint spec only if a parameter is not already defined in the constraint.
+//
+//	Returns a copy of the constraint with the defaults or an error in processing the schema and constraint.
+func (c *Client) withDefaultParams(constraint *unstructured.Unstructured, templ *templates.ConstraintTemplate) (*unstructured.Unstructured, error) {
+	if templ.Spec.CRD.Spec.Validation == nil || templ.Spec.CRD.Spec.Validation.LegacySchema != nil {
+		return constraint, nil
+	}
+
+	cpy := constraint.DeepCopy()
+	defaults := map[string]*apiextensions.JSON{}
+	schema := templ.Spec.CRD.Spec.Validation.OpenAPIV3Schema
+
+	for paramName := range schema.Properties {
+		paramProps := schema.Properties[paramName]
+
+		if paramProps.Default != nil {
+			defaults[paramName] = paramProps.Default
+		}
+	}
+
+	_, _, err := unstructured.NestedFieldNoCopy(cpy.Object, "spec", "parameters")
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", apiconstraints.ErrInvalidConstraint, err)
+	}
+
+	for paramName, defaultParam := range defaults {
```

Comment:
Does K8s have this logic already? Can we call that logic?

Reply:
At first glance, I did not find something to use in `apiextensions` but maybe I wasn't looking in the right place? I've looked at `SetDefaults_x` and how we have an existing wired up way to add some defaults in CF but I wasn't sure if they could be used for the defaults that I care about here.